tough/src/editor/targets.rs

```diff
@@ -0,0 +1,564 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT OR Apache-2.0
+#![allow(clippy::used_underscore_binding)] // #20
+
+//! Provides a `TargetsEditor` object for building and editing targets roles.
+
+use crate::editor::signed::{SignedDelegatedTargets, SignedRole};
+use crate::error::{self, Result};
+use crate::fetch::fetch_max_size;
+use crate::key_source::KeySource;
+use crate::schema::decoded::{Decoded, Hex};
+use crate::schema::key::Key;
+use crate::schema::{
+    DelegatedRole, DelegatedTargets, Delegations, KeyHolder, PathSet, RoleType, Signed, Target,
+    Targets,
+};
+use crate::transport::Transport;
+use crate::Limits;
+use crate::Repository;
+use chrono::{DateTime, Utc};
+use ring::rand::SystemRandom;
+use serde_json::Value;
+use snafu::{OptionExt, ResultExt};
+use std::borrow::Cow;
+use std::collections::HashMap;
+use std::num::NonZeroU64;
+use std::path::Path;
+use url::Url;
+
+const SPEC_VERSION: &str = "1.0.0";
+
+/// If you are not working with a repository that utilizes delegated targets, use the `RepositoryEditor`.
+///
+/// `TargetsEditor` contains the various bits of data needed to construct
+/// or edit a `Targets` role.
+///
+/// `TargetsEditor` should only be used with repositories that utilize delegated targets.
+/// `TargetsEditor` cannot create a `SignedRepository`. Whenever a user has access to `snapshot.json` and `timestamp.json` keys,
```

Comment:
Nit, this doesn't seemed to be wrapped at the width of the rest of the documentation.

```suggestion
/// `TargetsEditor` cannot create a `SignedRepository`. Whenever a user has access to `snapshot.json` and `timestamp.json` keys,
```